.github/workflows/release.yaml

```diff
@@ -46,15 +46,15 @@ jobs:
 
       - name: Checkout dist repo
         uses: actions/checkout@v2
-        if: steps.changesets.outputs.published
+        if: steps.changesets.outputs.published && steps.changesets.outputs.published != 'false'
```

Comment:
Maybe link to an issue if you file it?

Reply:
So, I thought this was an issue with their action but, after some more digging, I'm not so sure. AFAICT, action outputs are all supplied by writing to standard out in a special format, but the actual content of that output is just an arbitrary string.

Here's where the changesets action uses `@actions/core` to set an output:

https://github.com/changesets/action/blob/1ed9bc24b7a56462c183eb815c8f4bdf0e2e5785/src/index.ts#L91

... which calls:

https://github.com/actions/toolkit/blob/15e23998268e31520e3d93cbd106bd3228dea77f/packages/core/src/core.ts#L190-L193

... which calls:

https://github.com/actions/toolkit/blob/15e23998268e31520e3d93cbd106bd3228dea77f/packages/core/src/command.ts#L23-L30

This makes me think that the issue is really more that GitHub Actions i/o is all string based and not really related to the changeset action itself. Though, I did run across this comment which points out this [`fromJSON`](https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson) function as a pretty clean way of handling it:

https://github.community/t/boolean-evaluation-workflow-yaml/173565/2

Though, I'm not sure how `fromJSON` reacts to things that don't parse as JSON and I haven't found anything in the docs about that yet. WDYT? Should I switch over to `fromJSON`?

Reply:
I think I'd just keep what you have here, which seems like the simplest solution, add a comment about why, and file an issue on changesets to point out the gotcha. Their docs do say "A boolean value to indicate whether a publishing is happened or not", so at the very least the README should be updated, since it's not a boolean: https://github.com/changesets/action